model_utils/network.py

```diff
@@ -1,188 +1,322 @@
             input=input,
             size=size,
-            act=paddle.activation.Linear(),
+            act=None,
+            param_attr=fluid.ParamAttr(name=name + '_reverse_fc_weight'),
             bias_attr=False)
-        # batch norm is only performed on input-state projection
-        input_proj_bn_forward = paddle.layer.batch_norm(
-            input=input_proj_forward, act=paddle.activation.Linear())
-        input_proj_bn_backward = paddle.layer.batch_norm(
-            input=input_proj_backward, act=paddle.activation.Linear())
+        #batch norm is only performed on input-state projection
+        input_proj_bn_forward = fluid.layers.batch_norm(
+            input=input_proj_forward,
+            act=None,
+            param_attr=fluid.ParamAttr(
+                name=name + '_forward_batch_norm_weight'),
+            bias_attr=fluid.ParamAttr(name=name + '_forward_batch_norm_bias'),
+            moving_mean_name=name + '_forward_batch_norm_moving_mean',
+            moving_variance_name=name + '_forward_batch_norm_moving_variance')
+        input_proj_bn_backward = fluid.layers.batch_norm(
+            input=input_proj_backward,
+            act=None,
+            param_attr=fluid.ParamAttr(
+                name=name + '_reverse_batch_norm_weight'),
+            bias_attr=fluid.ParamAttr(name=name + '_reverse_batch_norm_bias'),
+            moving_mean_name=name + '_reverse_batch_norm_moving_mean',
+            moving_variance_name=name + '_reverse_batch_norm_moving_variance')
         # forward and backward in time
-        forward_simple_rnn = paddle.layer.recurrent(
-            input=input_proj_bn_forward, act=act, reverse=False)
-        backward_simple_rnn = paddle.layer.recurrent(
-            input=input_proj_bn_backward, act=act, reverse=True)
-
-    return paddle.layer.concat(input=[forward_simple_rnn, backward_simple_rnn])
+        forward_rnn = simple_rnn(
+            input=input_proj_bn_forward,
+            size=size,
+            param_attr=fluid.ParamAttr(name=name + '_forward_rnn_weight'),
+            bias_attr=fluid.ParamAttr(name=name + '_forward_rnn_bias'),
+            is_reverse=False)
+        reverse_rnn = simple_rnn(
+            input=input_proj_bn_backward,
+            size=size,
+            param_attr=fluid.ParamAttr(name=name + '_reverse_rnn_weight'),
+            bias_attr=fluid.ParamAttr(name=name + '_reverse_rnn_bias'),
+            is_reverse=True)
+    out = fluid.layers.concat(input=[forward_rnn, reverse_rnn], axis=1)
+    return out
 
 
 def bidirectional_gru_bn_layer(name, input, size, act):
     """Bidirectonal gru layer with sequence-wise batch normalization.
     The batch normalization is only performed on input-state weights.
-
     :param name: Name of the layer.
     :type name: string
     :param input: Input layer.
-    :type input: LayerOutput
-    :param size: Number of RNN cells.
+    :type input: Variable
+    :param size: Number of GRU cells.
```

Comment:
`dimension of GRU cells`. And pay attention to this arguments in other places.

Reply:
Done